Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -275,6 +275,48 @@ def check_python_symlinks(name, keg_only)
     "Python formulae that are keg-only should not create `pip3` and `wheel3` symlinks."
   end
 
+  INVALID_EMPTY_FILE_NAMES = %w[AUTHORS ChangeLog changes COPYING LICENSE NEWS README TODO].freeze
+  INVALID_EMPTY_FILE_EXTENSIONS = %w[.html .md .rst .txt].freeze
+  INVALID_EMPTY_FILE_DIRECTORIES = %w[bin libexec share/man share/doc].freeze
+  # Names and extensions to ignore, even if they are in a location within prefix given above
+  VALID_EMPTY_FILE_NAMES = %w[__init__ empty REQUESTED].freeze
+  VALID_EMPTY_FILE_EXTENSIONS = %w[.gitkeep .keep .keepme .stamp].freeze
+
+  def check_empty_files(prefix)
+    return unless prefix.directory?
+
+    empty_files = prefix.children.select do |f|
+      next if INVALID_EMPTY_FILE_NAMES.none?(File.basename(f)) &&
+              INVALID_EMPTY_FILE_EXTENSIONS.none?(File.extname(f))
+      next unless File.zero?(f)
+
+      true
+    end
+
+    INVALID_EMPTY_FILE_DIRECTORIES.each do |dir|
+      path = prefix/dir
+      next unless path.directory?
+
+      # search all directories in INVALID_EMPTY_FILE_LOCATIONS and their subdirectories
+      Pathname.glob(path/"**/*").each do |p|
+        # Any files in the "valid" list should be skipped because they are expected to be empty
+        next if VALID_EMPTY_FILE_NAMES.include?(p.basename(".*").to_s.delete_suffix("/"))
```

Comment:
Why is the `delete_suffix` needed when using `p.basename` compared to `File.basename`?

Reply:
Okay, I think it's not actually needed. I originally did that because using `p.basename` failed (`include?` was always `false` regardless of the name), while `File.basename` had correct behavior. I tried using print statements and saw `/`'s at the end of `p.basename`, so I tried to remove it.

Apparently it does work correctly if I simply use `p.basename(".*").to_s`, without deleting a suffix; I simply made both changes at once and assumed the `delete_suffix` call was needed. I'll remove it, thanks.

Reply:
👍🏻 🆒

Reply:
Also, just to check: is `p.basename(".*")` definitely required over `p.basename`? What cases does the prior catch that the latter does not?

Reply:
`p.basename(".*")` excludes the extension. For example, if I have a file called `README.old` (example):

- `p.basename(".*")` returns just `README`
- `p.basename` returns `README.old`

Thus only `p.basename` does not match anything in the file name array. It also won't match anything in the extension array (since `.old` is not there), and thus this file would've otherwise been missed.

Reply:
Gotcha, thanks!